src/gpu/optixPrograms.cu

```diff
@@ -203,6 +203,7 @@ extern "C" __global__ void __closesthit__()
 
 		intensity = tex2D<TextureTexelFormat>(entityData.texture, uv[0], uv[1]);
 	}
+	intensity *= cosf(incidentAngle);
```

Comment:
@msz-rai  Isn't this redundant? The absolute value of the dot product is already known, no need to calculate cos on an acos output. The above incident angle calculation has to be simply split into two statements. One to calculate and store the absolute value of the dot product (this value can be used here). Second one for calculation of the incident angle. To the best of my knowledge tryg functions are rather expensive (i would guess that even more so on the gpu).